internal/consul/certmanager.go

```diff
@@ -236,34 +291,16 @@ func (c *CertManager) Manage(ctx context.Context) error {
 	c.leafWatch.HybridHandler = c.handleLeafWatch
```

Comment:
To reimplement this without relying on the Consul agent, we'd basically need to rewrite all of the logic from https://github.com/hashicorp/consul/blob/main/agent/cache-types/connect_ca_leaf.go#L514-L685 to generate a new leaf certificate (which currently depends on internal methods of Consul's `connect` package) and as well as tie it into the CA root watch handler to regenerate certificates on rotation, as the gRPC API only exposes the lower-level `Sign` method.